examples/GANwithSelf-taughtLearning/GAN/lpips/__init__.py

```diff
@@ -0,0 +1,168 @@
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import skimage
+import torch
+from torch.autograd import Variable
+
+from lpips import dist_model
+
+
+if skimage.__version__ == '0.14.3':
+    from skimage.measure import compare_ssim
+else:
+    from skimage.metrics import structural_similarity as compare_ssim
+
+
+
+class PerceptualLoss(torch.nn.Module):
+    def __init__(self, model='net-lin', net='alex', colorspace='rgb', spatial=False, use_gpu=True, gpu_ids=[0]): # VGG using our perceptually-learned weights (LPIPS metric)
+    # def __init__(self, model='net', net='vgg', use_gpu=True): # "default" way of using VGG as a perceptual loss
```

Comment:
if this comment of code is still useful, please write a more detailed comment
if not useful, please remove it